### PR TITLE
DataTable

### PR DESCRIPTION
**Pull request by cheziyi**

### CODE REVIEW

**Comment by cheziyi** [comment]
I expect to modify this slightly when updating Core, depending on any additional necessary properties.

File: PrismaDB-QueryAST/Result/DataColumn.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using PrismaDB.QueryAST.DDL;
+using PrismaDB.QueryAST.DML;
+
+namespace PrismaDB.QueryAST.Result
+{
+    public class DataColumn
```
Comment (bazzilic) [inline PrismaDB-QueryAST/Result/DataColumn.cs]: Maybe better to call `DataColumnHeader` or `ResultColumnHeader`. I thought you are organizing data by columns initially when I saw the class name.

File: PrismaDB-QueryAST/Result/DataTable.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using PrismaDB.Commons;
+using PrismaDB.QueryAST.DML;
+
+namespace PrismaDB.QueryAST.Result
+{
+    public class DataTable
```
Comment (bazzilic) [inline PrismaDB-QueryAST/Result/DataTable.cs]: Same here.

File: PrismaDB-QueryAST/DML/Expressions/Expression.cs [hunk 3]
```diff
@@ -14,10 +14,12 @@ public abstract class Expression : ICloneable
 
         public abstract object Clone();
 
-        public abstract object Eval(DataRow r);
+        public abstract object Eval(ResultRow r);
 
         public abstract List<ColumnRef> GetColumns();
 
+        public abstract string DisplayName();
```
[the remark this answers is not in the document]
Comment (cheziyi) [reply PrismaDB-QueryAST/DML/Expressions/Expression.cs]: Removed this, just had a method for ColumnRef only.

**Comment by bazzilic** [comment]
All looks alright to, so you can merge when ready

**Pull request merged**